In [1]:
import pandas as pd
from datetime import datetime as dt

In [4]:
#temp observations
df = pd.read_csv('./data/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [7]:
df.rename(columns={"station": "Station", "date": "Date", "prcp": "Precipitation", 
                   "tobs": "Temp_observations"}, inplace=True)
df.head()

Station        Date  Precipitation  Temp_observations
0  USC00519397  2010-01-01           0.08                 65
1  USC00519397  2010-01-02           0.00                 63
2  USC00519397  2010-01-03           0.00                 74
3  USC00519397  2010-01-04           0.00                 76
4  USC00519397  2010-01-06            NaN                 73

In [9]:
df.Date.dtype

dtype('O')

In [11]:
#Convert the date column format from string to datetime
df.Date = pd.to_datetime(df.Date, infer_datetime_format=True)

In [12]:
df = df.set_index(df['Date'])

In [13]:
df = df.drop(columns= 'Date')
df.head()

Station  Precipitation  Temp_observations
Date                                                     
2010-01-01  USC00519397           0.08                 65
2010-01-02  USC00519397           0.00                 63
2010-01-03  USC00519397           0.00                 74
2010-01-04  USC00519397           0.00                 76
2010-01-06  USC00519397            NaN                 73

In [15]:
from scipy import stats

In [16]:
june_df = df[df.index.month == 6]
dec_df = df[df.index.month == 12]

In [17]:
june_df.mean()

Precipitation         0.136360
Temp_observations    74.944118
dtype: float64

In [18]:
june_temp = june_df.Temp_observations
dec_temp = dec_df.Temp_observations

In [19]:
stats.ttest_ind(june_temp, dec_temp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

Analysis:
Across all the stations, the mean temperatures in June and December temperature in years 2010-2017 differ by 3.9 degrees Celsius.  An unpaired t-test was conducted, and with an extremely low p-value, the difference is deemed statistically significant.